File: python3/easy/power-of-thor.py
lx, ly, tx, ty = [int(i) for i in input().split()]

while True:
    _ = int(input())

    if ly > ty:
        move = "S"
        ty += 1
    elif ly < ty:
        move = "N"
        ty -= 1
    else:
        move = ""

    if lx > tx:
        move += "E"
        tx += 1
    elif lx < tx:
        move += "W"
        tx -= 1

    print(move)

File: javascript/easy/power-of-thor.js
[lx, ly, tx, ty] = readline().split(' ').map(n => + n);

while (true) {
    + readline();
    
    var move = '';
    if (ly > ty) {
        move = 'S';
        ty++;
    } else if (ly < ty) {
        move = 'N';
        ty--;
    }

    if (lx > tx) {
        move += 'E';
        tx++;
    } else if (lx < tx) {
        move += 'W';
        tx--;
    }
    print(move);
}
